File: sol/3228/3228.cpp
struct Solution {
    int maxOperations(string_view s) {
        int res = 0, count = 0, total = 0;
        char prev = ' ';
        for (char c : s){
            if (c != prev){
                if (prev == '1'){
                    total += count;
                    res += total;
                }
                count = 0;
                prev = c;
            }
            if (c == '1'){
                ++count;
            }
        }
        return res;
    }
};

File: sol/3228/3228.py
class Solution:
    def maxOperations(self, s: str) -> int:
        res = 0
        count = 0
        total = 0
        prev = ' '
        for x in s:
            if x != prev:
                if prev == '1':
                    total += count
                    res += total
                count = 0
                prev = x
            if x == '1':
                count += 1
        return res
